webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -651,54 +652,155 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
         int maxMoves = WCA_MAX_MOVES_FMC;
-
+        
         HashMap<String, String> substitutions = new HashMap<String, String>();
         substitutions.put("maxMoves", ""+maxMoves);
-        rulesList.add("• "+translate("fmc.rule4", locale, substitutions));
-
-        rulesList.add("• "+translate("fmc.rule5", locale));
-        rulesList.add("• "+translate("fmc.rule6", locale));
-        rulesList.add("  • "+translate("fmc.faceMoves", locale));
-        rulesList.add("    • "+String.format("R     U     F     L     B     D      (%s)", translate("fmc.clockwise", locale)));
-        rulesList.add("    • "+String.format("R'    U'    F'    L'    B'    D'      (%s)", translate("fmc.counterClockwise", locale)));
-        rulesList.add("    • "+String.format("R2   U2   F2   L2   B2   D2    (%s)", translate("fmc.double", locale)));
-        rulesList.add("  • "+translate("fmc.rotations", locale));
-        rulesList.add("    • "+String.format("[r]     [u]     [f]     [l]     [b]     [d]      (%s)", translate("fmc.clockwise", locale)));
-        rulesList.add("    • "+String.format("[r']    [u']    [f']    [l']    [b']    [d']      (%s)", translate("fmc.counterClockwise", locale)));
-        rulesList.add("    • "+String.format("[r2]   [u2]   [f2]   [l2]   [b2]   [d2]    (%s)", translate("fmc.double", locale)));
+        rulesList.add(" •  "+translate("fmc.rule4", locale, substitutions));
+        
+        rulesList.add(" •  "+translate("fmc.rule5", locale));
+        rulesList.add(" •  "+translate("fmc.rule6", locale));
 
         int rulesTop = competitorInfoBottom + (withScramble ? 65 : 153);
-        Rectangle rulesRectangle = new Rectangle(left+padding, scrambleBorderTop+padding*2, competitorInfoLeft-padding, rulesTop);
 
+        Rectangle rulesRectangle = new Rectangle(left+padding, scrambleBorderTop+tableHeight, competitorInfoLeft-padding, rulesTop);
         String rules = String.join("\n", rulesList);
-        float leadingMultiplier = 1.5f; // default pdf leading
-
-        float potentialFontSize = fitText(new Font(bf), rules, rulesRectangle, 15f, true, leadingMultiplier);
-
-        ColumnText ct = new ColumnText(cb);
-        ct.setSimpleColumn(rulesRectangle);
-        Paragraph paragraph = new Paragraph(rules);
-        paragraph.setFont(new Font(bf, potentialFontSize));
-        paragraph.setMultipliedLeading(leadingMultiplier);
-        ct.addElement(paragraph);
-
-        ct.go();
-
+        fitAndShowTextNew(cb, rules, bf, rulesRectangle, 15, Element.ALIGN_LEFT, 1.5f);
+        
         doc.newPage();
     }
-
+    
+    private static void fitAndShowTextNew(PdfContentByte cb, String text, BaseFont bf, Rectangle rect, float maxFontSize, int align, float leading) throws DocumentException {
+        // We create a temp pdf and check if the text fit in a rectangle there.
+        // If it's ok, we add the text to original pdf.
+        
+        //TODO replace fitAndShowText (old) with this new one
```

Comment:
Can you make an issue for this and update this comment to link to the the issue?

Reply:
Done.